### ISSUE TITLE
cmd/compile: large constant strings do not get reused

### ISSUE THREAD

**Issue by mpvl** [golang/go#13145]
In x/text/language/tables.go, if I replace the string const types with var, I get a significant reduction in binary size for:

```go
package main
import (
	"fmt"
	"golang.org/x/text/language"
)
func main() {
	fmt.Print(language.Version)
}
```

**Comment by josharian** [comment]
@mpvl a fix went into 1.7 that should have helped with this. Can you confirm/disconfirm?

**Comment by mpvl** [comment]
I can no longer reproduce it. It appears to be fixed! (In fact, the binary gets slightly smaller now switching to string consts.)  

Thanks!
